huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go

```diff
@@ -326,6 +340,17 @@ func resourceFgsFunctionV2Create(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if strNum, ok := d.GetOk("max_instance_num"); ok {
+		maxInstanceNum, err := strconv.Atoi(strNum.(string))
+		if err != nil {
```

Comment:
it's better to convert the value to int before creating the FGS function.

Reply:
bring this type conversion (validation) into the schema definition (ValidateFunc).